core/src/main/java/org/apache/calcite/adapter/enumerable/RexImpTable.java

```diff
@@ -327,6 +328,7 @@
     defineMethod(SOUNDEX, BuiltInMethod.SOUNDEX.method, NullPolicy.STRICT);
     defineMethod(DIFFERENCE, BuiltInMethod.DIFFERENCE.method, NullPolicy.STRICT);
     defineMethod(REVERSE, BuiltInMethod.REVERSE.method, NullPolicy.STRICT);
+    defineMethod(IF, BuiltInMethod.IF.method, NullPolicy.STRICT);
```

Comment:
The `NullPolicy` should be `NONE`, and you should handle the cases with `nul`l in `SqlFunctions.ifFunction`.